src/openforms/submissions/api/permissions.py

```diff
@@ -81,30 +81,16 @@ def filter_queryset(self, request: Request, view: APIView, queryset):
         return queryset.filter(uuid__in=active_submissions)
 
 
-class OwnsTemporaryUploadPermission(permissions.BasePermission):
+class OwnsTemporaryUploadPermission(AnyActiveSubmissionPermission):
     """
     Verify the upload is registered in the users session
     """
 
-    def has_permission(self, request: Request, view: APIView) -> bool:
-        active_uploads = request.session.get(UPLOADS_SESSION_KEY)
-        if not active_uploads:
-            return False
-        return True
-
-    def has_object_permission(self, request: Request, view: APIView, obj) -> bool:
-        active_uploads = request.session.get(UPLOADS_SESSION_KEY)
-
-        upload_url_kwarg = view.lookup_url_kwarg or view.lookup_field
-        upload_uuid = view.kwargs[upload_url_kwarg]
-
-        return str(upload_uuid) in active_uploads
-
-    def filter_queryset(self, request: Request, view: APIView, queryset):
-        active_uploads = request.session.get(UPLOADS_SESSION_KEY)
-        if not active_uploads:
-            return queryset.none()
-        return queryset.filter(uuid__in=active_uploads)
+    def has_object_permission(
+        self, request: Request, view: APIView, obj: TemporaryFileUpload
+    ) -> bool:
+        submission_uuid = str(obj.submission.uuid)
+        return owns_submission(request, submission_uuid)
```

Comment:
I'm wondering if `super().has_object_permission` should be called just for safety? Currently `AnyActiveSubmissionPermission` doesn't override it, but it might in the future?

Reply:
that doesn't make sense? the scope of the super class is that you are allowed to proceed as soon as you have started any submission at all, there are no object level permissions in play there. And, there is no guarantee that IF they were to be added, the objects will share a common interface, they can be completely different models with a completely different relation to a submission.